### PR TITLE
Update Github issue templates to use issue forms

### LINKED ISSUE #1831: [FEATURE REQUEST] Use yml and incorporate other form elements in issue templates
### Name and Institution (Required)

Name: Lizzie Lundgren
Institution: Harvard University

### New GEOS-Chem feature or discussion

This is non-urgent nice-to-have feature request for a rainy day.

Sometimes a submitted GitHub issue does not have all of the information that GCST requests, or has a format that is difficult to read. We could encourage better issue reports by using drop-down menus and comment boxes in our issue templates. To facilitate this we could also change from using markdown to yaml format in the template files.

An example of another model that does this is CAM in CESM. See [here](https://github.com/ESCOMP/CAM/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml&title=Put+brief+%28%3C65+char%29+bug+title+here%21) for what the user sees and [here](https://github.com/ESCOMP/CAM/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml?plain=1) for the template code.

GitHub documentation on this topic::
[Syntax for issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)
[Syntax for GitHub's form schema](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema)

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Describe the update

Github issue template files are updated to utilize [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). This involves using YAML configuration files instead of markdown. This should make issue submissions easier for users and will ensure users provide the required information.
    
Note that issue forms are not supported for pull requests currently. 
    
### Expected changes

This is a zero difference update to model output. It will only impact the look of submitting issues on Github once merged into the `main` branch.

### Related Github Issue(s)

- Closes #1831
- https://github.com/geoschem/HEMCO/pull/276
- https://github.com/geoschem/GCClassic/pull/58
- https://github.com/geoschem/GCHP/pull/415

### CODE REVIEW

**Comment by msulprizio** [comment]
A temporary repository has been set up at https://github.com/msulprizio/test_issues_form/issues/new/choose for testing these new forms without merging into `main` here.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    attributes:
+      label: What are the steps to reproduce the bug?
+      placeholder: Explain what you did
+    validations:
+      required: true
+  - type: textarea
+    id: log-files
+    attributes:
+      label: Please attach any relevant configuration and log files.
+      description: Drag and drop files like the GEOS-Chem log file, geoschem_config.rc, and HEMCO_Config.rc here. You may need to rename files to end in .txt first.
```
Comment (yantosca) [inline .github/ISSUE_TEMPLATE/bug-report.yml:36]: Typo: `geoschem_config.rc` -> `geoschem_config.yml`

Also add `HISTORY.rc` here

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      required: true
+  - type: dropdown
+    id: compiler
+    attributes:
+      label: What compiler were you using?
+      multiple: true
+      options:
+        - Intel
+        - GNU
+        - PGI
```
Comment (yantosca) [inline .github/ISSUE_TEMPLATE/bug-report.yml:66]: You may want to say `PGI (GEOS-Chem in CESM only)`

@lizziel would know the other compilers used by CESM

Comment (yantosca) [reply .github/ISSUE_TEMPLATE/bug-report.yml:66]: Also would be nice to have a text box where they can type in the compiler version.

Comment (yantosca) [reply .github/ISSUE_TEMPLATE/bug-report.yml:66]: I would also list GNU first, so that they are in alphabetical order

**Review by yantosca — CHANGES_REQUESTED** [review]
Thanks @msulprizio.  This looks great overall, I've just noted a few minor things.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    attributes:
+      label: What are the steps to reproduce the bug?
+      placeholder: Explain what you did
+    validations:
+      required: true
+  - type: textarea
+    id: log-files
+    attributes:
+	label: Please attach any relevant configuration and log files.
+	description: Drag and drop files like the GEOS-Chem log file, geoschem_config.rc, and HEMCO_Config.rc here. You may need to rename files to end in .txt first.
```
[the remark this answers is not in the document]
Comment (msulprizio) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: `build_info/summarize_build` has now been added to the list of files to provided

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (msulprizio) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: Fixed typo and added `HISTORY.rc` to the list of files to provided in commit https://github.com/geoschem/geos-chem/pull/2288/commits/e4a555f0d86ea247dfb19883ed04850304a322b4

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (msulprizio) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: In commit https://github.com/geoschem/geos-chem/pull/2288/commits/9a12d4819a4235dce35a0b7267eaac0e33f2727a, I removed the compiler dropdown and instead provide a text box for users to specify the compiler and version used

Screenshot 2024-05-21 at 2 18 07 PM

Comment (lizziel) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: We are referring CESM users to report issues at the CESM forum. I think we should limit the compilers listed to be for the offline model.

**Review by lizziel — CHANGES_REQUESTED** [review]
This looks great for GC-Classic issues. I wonder if we should have a drop-down for what simulation, e.g. fullchem, and configuration, e.g. GC-Classic nested, they are using. Sometimes GCHP users create issues on GEOS-Chem repo but there is no place for them to say they are using GCHP.

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (yantosca) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: One small detail, summarize_build is a script.  Maybe say "the output from running build_info/summarize_build" or something like that.  Sorry for the nitpick.

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (yantosca) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: The text box is fine.  Thanks @msulprizio

**Review by yantosca — APPROVED** [review]
Thanks for the updates @msulprizio!  Good to merge, pending @lizziel's additional comments

**Comment by msulprizio** [comment]
> I wonder if we should have a drop-down for what simulation, e.g. fullchem, and configuration, e.g. GC-Classic nested, they are using. Sometimes GCHP users create issues on GEOS-Chem repo but there is no place for them to say they are using GCHP.

@lizziel That's a good point! I added additional entries to the form for configuration, simulation type, resolution, and met field. 

Screenshot 2024-05-21 at 3 44 41 PM

Let me know if you think anything else should be added/changed!

**Comment by yantosca** [comment]
@lizziel @msulprizio, this looks great!

**Comment by lizziel** [comment]
Looks good! A few more comments: 
- If using GCHP then users should also provide MPI library and version
- If we include CESM and WRF-GC then we should also include GEOS
- I'm hesitant to start down the path of asking for libraries and config files for GEOS-Chem in external models. GCHP run issues should be in the GCHP github, and GEOS-Chem in GEOS issues go to the GMAO github, and CESM issues go to the CESM user forum. I don't think we give WRF-GC help either. I wonder if we should add a note pointing people to those resources?

**Comment by lizziel** [comment]
Thinking more about this, I can see GMAO developers making an issue (although write now they just ping me on their own github issues) and NCAR developers too if they have a question or something breaks. But I still wonder if we should list other resources somewhere such as GCHP GitHub for GCHP issues, or explain GEOS-Chem repo issues are for all GC-Classic issues plus science codebase issues if run in GCHP or other models.

**Review by lizziel — APPROVED** [review]

**Comment by msulprizio** [comment]
@lizziel I removed the options related to external models from the configuration question and added a note for users to check with the external model's support.

**Pull request merged**